### ISSUE TITLE
no-relative-parent-imports reports internal packages as parent

### ISSUE THREAD

**Issue by JacobLey** [import-js/eslint-plugin-import#2467]
I recently started working on a rush monorepo, that basically has the setup:

```
rush.json
/lib
/lib/src/index.js -- export const abc = 123
/apps
/apps/src/index.js -- import { abc } from '@my-package/lib';
/apps/node_modules/@my-package/lib/src/index.js
```

I have eslint settings configured as:

```
settings: {
    'import/internal-regex': '^@my-package/',
},
```

The rule `import/no-relative-parent-imports` wrongly reports that `import { abc } from '@my-package/lib'` with error `Relative imports from parent directories are not allowed. Please either pass what you're importing through at runtime (dependency injection), move `index.ts` to same directory as `@my-package/lib` or consider making `@my-package/lib` a package.`

I suspect this is related to https://github.com/import-js/eslint-plugin-import/issues/1644

Upon some digging, it appears that this library _does_ correctly identify that import as "internal" (therefore rules like `import/order` work as expected) but then the path is "resolved" to `../node_modules/@my-package/lib/index.js` which looks like a parent directory.

[https://github.com/import-js/eslint-plugin-import/blob/main/src/rules/no-relative-parent-imports.js#L24] This check returns false because it is an `internal` import, not `export`
[`absDepthPath`](https://github.com/import-js/eslint-plugin-import/blob/main/src/rules/no-relative-parent-imports.js#L28) resolves to the node_modules path then [`relDepthPath`](https://github.com/import-js/eslint-plugin-import/blob/main/src/rules/no-relative-parent-imports.js#L34) looks like a [parent import](https://github.com/import-js/eslint-plugin-import/blob/main/src/rules/no-relative-parent-imports.js#L36) when converted to a relative path

I suggestion is to simply exit early on both `internal` and `builtin` (because why not?) modules instead of trying to resolve the path. Is there a reason it was not implemented that way in the first place? This library has generalized logic for detecting "parent" vs "sibling" logic already?

It is worth noting that I worked with this rule in an npm/lerna workspace and never came across this issue. So I tried investigating those and it appears in those workspaces, `node_modules` was hoisted to the top-most directory and it appears [`resolve`](https://github.com/import-js/eslint-plugin-import/blob/main/utils/resolve.js#L217) actually [failed to resolve](https://github.com/import-js/eslint-plugin-import/blob/main/src/rules/no-relative-parent-imports.js#L30) those and it would not throw an error.

So in that sense it seems that Rush is actually providing a _better_ node_modules resolution, and as a consequence these rules which might have allowed internal packages to slip by are now exposing a logical flaw...

Happy to open the PR to allow `internal` + `builtin` modules to exit early, but I suspect I am missing a piece of the logic, so opening this issue for discussion!

**Comment by ljharb** [comment]
While a monorepo blindly hoisting things to the top level is always going to be problematic, I agree that anything that's inside a parent's `node_modules` directory should not be warned on by this rule.

That a module is marked "internal" or "builtin" doesn't seem relevant.

**Comment by tmtron** [comment]
@ljharb I also think that internal modules should  be ignored.

But maybe it's just a misunderstanding, as I am not sure what "internal" means in this context.
So I'll try to explain what I think that the terms mean to me.

I have created a minimal nx monorepo  on GitHub to illustrate the issue: [GitHub lint-test](https://github.com/tmtron/lint-test).  

* **external**: listed in [package.json](https://github.com/tmtron/lint-test/blob/master/package.json#L13) and installed in the `node_modules` folder, e.g.
  * `import from 'lodash';`
  * `import from '@angular/common';`
* **internal**: in a monorepo, we can have multiple packages/libs
  * these libs are checked-into the VCS and the [nrwl/nx](https://nx.dev/) makes sure to build them in the right order, etc.
  * these imports look the same as external ones, but they are specified in the path mappings of [tsconfig.json](https://github.com/tmtron/lint-test/blob/26820ec383984e41319876a68b6aaffa0a33d83c/tsconfig.base.json#L18)
    * `"@lint-test/lib-a": ["libs/lib-a/src/index.ts"]`
    * `"@lint-test/lib-b": ["libs/lib-b/src/index.ts"]`
    * these can be imported like this: `import { libA } from '@lint-test/lib-a';`  [ref](https://github.com/tmtron/lint-test/blob/26820ec383984e41319876a68b6aaffa0a33d83c/libs/lib-b/src/lib/childdir/child.ts#L6)
* **builtin**: e.g. `import from 'fs'` (i.e. node:fs)
  * see: https://github.com/import-js/eslint-plugin-import#importcore-modules

As explained in the readme, when [lib-b](https://github.com/tmtron/lint-test/tree/master/libs/lib-b) uses `import { libA } from '@lint-test/lib-a'`, I am sure, that this should **not** be an error.
But currently `no-relative-parent-imports` flags them as error: [see import in lib-a child.tsref](https://github.com/tmtron/lint-test/blob/26820ec383984e41319876a68b6aaffa0a33d83c/libs/lib-b/src/lib/childdir/child.ts#L6)

Am I missing something?

**Comment by trim21** [comment]
I write a simple rule to fix this, you will need [eslint-plugin-rulesdir](https://www.npmjs.com/package/eslint-plugin-rulesdir)

```js
// remove this after https://github.com/import-js/eslint-plugin-import/issues/2467 is fixed

module.exports = {
  meta: {
    type: 'problem',
    docs: {
      description: 'forbidden relative parent import',
      category: 'Stylistic Issues',
    },
    schema: [],
  },

  create: (context) => {
    return {
      'Program > ImportDeclaration': (node) => {
        if (node.source.value.startsWith('../')) {
          context.report({
            node,
            message: 'do not use relative parent import, use absolute import ...',
          });
        }
      },
    };
  },
};
```
